core/templates/dev/head/pages/exploration-player-page/learner-experience/conversation-skin.directive.ts

```diff
@@ -724,45 +727,52 @@ angular.module('oppia').directive('conversationSkin', [
                 var storyId = UrlService.getUrlParams().story_id;
                 var nodeId = UrlService.getUrlParams().node_id;
                 $scope.inStoryMode = true;
-                $scope.storyViewerUrl = '/story/' + storyId;
-              }
-
-              ExplorationRecommendationsService.getRecommendedSummaryDicts(
-                recommendedExplorationIds,
-                includeAutogeneratedRecommendations,
-                function(summaries) {
-                  $scope.recommendedExplorationSummaries = summaries;
-                  if ($scope.inStoryMode) {
-                    StoryViewerBackendApiService.fetchStoryData(storyId).then(
-                      function(storyDataDict) {
-                        var storyNodes = storyDataDict.story_nodes.map(
-                          function(storyNodeDict) {
-                            return ReadOnlyStoryNodeObjectFactory
-                              .createFromBackendDict(storyNodeDict);
-                          });
-                        var completedStoryNodes = [];
-                        for (var i = 0; i < storyNodes.length; i++) {
-                          if (storyNodes[i].isCompleted()) {
-                            completedStoryNodes.push(storyNodes[i]);
-                          } else {
-                            $scope.nextNodeId = storyNodes[i].getId();
-                            break;
+                $scope.storyViewerUrl = UrlInterpolationService.interpolateUrl(
+                  STORY_VIEWER_URL_TEMPLATE, {
+                    story_id: storyId
+                  });
+                ExplorationRecommendationsService.
+                  recordCompletionAndGetNextExplorationSummary(
```

Comment:
If the arg that comes back from this is "summaries" (plural), should the function name be titled "...Summaries"?

Reply:
Renamed function.